File: download_logos.py
import os
import requests
from duckduckgo_search import DDGS

os.makedirs("public/images/clients", exist_ok=True)

queries = {
    "slb": "Schlumberger SLB logo transparent",
    "koc": "Kuwait Oil Company KOC logo transparent",
    "baker_hughes": "Baker Hughes logo transparent",
    "al_mansoori": "Al Mansoori Specialized Engineering logo transparent",
}

ddgs = DDGS()

for name, query in queries.items():
    print(f"Searching for {name}...")
    try:
        results = list(ddgs.images(query, max_results=5))
        for res in results:
            url = res['image']
            if not url.endswith((".png", ".jpg", ".jpeg", ".svg")):
                continue
            ext = ".png" if ".png" in url else ".jpg"
            print(f"Trying to download {url}...")
            try:
                r = requests.get(url, timeout=10, headers={'User-Agent': 'Mozilla/5.0'})
                if r.status_code == 200:
                    with open(f"public/images/clients/{name}{ext}", "wb") as f:
                        f.write(r.content)
                    print(f"Saved {name}{ext}")
                    break
            except Exception as e:
                print(f"Failed to download from {url}: {e}")
    except Exception as e:
        print(f"Failed to search for {name}: {e}")

File: download_logos.js
const fs = require('fs');
const https = require('https');
const path = require('path');

const dir = path.join(__dirname, 'public', 'images', 'clients');
if (!fs.existsSync(dir)) fs.mkdirSync(dir, { recursive: true });

const download = (url, dest) => {
  return new Promise((resolve, reject) => {
    https.get(url, (response) => {
      if (response.statusCode === 200) {
        const file = fs.createWriteStream(dest);
        response.pipe(file);
        file.on('finish', () => {
          file.close();
          resolve(true);
        });
      } else if (response.statusCode === 301 || response.statusCode === 302) {
        download(response.headers.location, dest).then(resolve).catch(reject);
      } else {
        console.error(`Status ${response.statusCode} for ${url}`);
        resolve(false);
      }
    }).on('error', (err) => {
      reject(err);
    });
  });
};

const domains = {
  slb: 'slb.com',
  koc: 'kockw.com',
  baker_hughes: 'bakerhughes.com',
  al_mansoori: 'almansoori.biz'
};

async function main() {
  for (const [name, domain] of Object.entries(domains)) {
    console.log(`Downloading ${name} from ${domain}...`);
    try {
      const success = await download(`https://logo.clearbit.com/${domain}?size=400`, path.join(dir, `${name}.png`));
      if (!success) console.log(`Failed to download ${name}`);
    } catch (e) {
      console.error(`Failed ${name}: ${e.message}`);
    }
  }
}
main();
